pilot/pkg/security/model/authentication.go

```diff
@@ -76,12 +86,19 @@ func ConstructSdsSecretConfigForCredential(name string) *tls.SdsSecretConfig {
 		return nil
 	}
 
+	if features.EnableSDSServer {
+		return &tls.SdsSecretConfig{
```

Comment:
isn't this also affecting sidecar (UpstreamClusterTLSContext from DestinationRule)? I thought this should only for gateway.

Reply:
No, because of https://github.com/istio/istio/blob/5917679c356f211299329fa09bedb5bebf41389f/pilot/pkg/networking/core/v1alpha3/cluster.go#L878-L884 (this is the existing code).

It *will* likely impact sidecar in the future, but for now we haven't fully designed that solution